### PR TITLE
Handle categories page with delete button

### PR DESCRIPTION
**Pull request by Tasadan-Filip**

### CODE REVIEW

File: src/app/components/categories-table/categories-table.component.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+@Component({
+  selector: 'app-categories-table',
+  templateUrl: './categories-table.component.html',
+  styleUrls: ['./categories-table.component.scss']
+})
+export class CategoriesTableComponent implements OnInit {
+
+  dataSource: Category[] = [];
+  categoriesToShowArray: CategoryToShow[] = [];
```
Comment (larisaChh) [inline src/app/components/categories-table/categories-table.component.ts]: Maybe "displayedCategories" sounds better?

File: src/app/components/categories-table/categories-table.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (larisaChh) [inline src/app/components/categories-table/categories-table.component.spec.ts]: Remove this file, we do not need it

File: src/app/components/categories-table/categories-table.component.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        defaultBackground: this.sanitizer.bypassSecurityTrustResourceUrl(category.defaultBackground),
+      }
+      this.categoriesToShowArray.push(newCategory);
+    }
+  }
+
+  deleteCategory(categoryId: number): void {
+    this.openChangeRoleDialog().subscribe(result => {
+      if (result)
+        this.endpointsService.deleteCategory(categoryId).subscribe(() => { this.categoriesToShowArray = this.categoriesToShowArray.filter((category: CategoryToShow) => category.id != categoryId) });
```
Comment (larisaChh) [inline src/app/components/categories-table/categories-table.component.ts]: Splitting such a long line like this helps readability

File: src/app/services/endpoints.service.ts [hunk 4]
```diff
@@ -26,4 +27,12 @@ export class EndpointsService {
...
     return this.http.get<Employee[]>(environment.apiUrl + "employee")
   }
+
+  getCategories(): Observable<CategoriesResponse> {
+    return this.http.get<CategoriesResponse>(environment.apiUrl + "Category") 
+  }
+  
+  deleteCategory(categoryId: number): Observable<unknown> {
+    return this.http.delete(environment.apiUrl + "Category" + "?categoryId=" + categoryId.toString());
+  }
```
Comment (larisaChh) [inline src/app/services/endpoints.service.ts:37]: You can also try using "params" to send queryParams :)
`return this.http.delete(environment.apiUrl + `Category`, {
            params: {
                categoryId: categoryId.toString()
            }
        });`

File: src/app/components/categories-table/categories-table.component.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
+import { Component, OnInit } from '@angular/core';
+import { Category, CategoryToShow } from 'src/app/models/category';
+import { EndpointsService } from 'src/app/services/endpoints.service';
+import { DomSanitizer } from '@angular/platform-browser';
+import { ConfirmComponent } from '../dialogs/confirm/confirm.component';
+import { Observable } from 'rxjs';
+import { DialogService } from 'src/app/services/dialog.service';
+import { MatTableModule } from '@angular/material/table';
```
Comment (meda-racasan) [inline src/app/components/categories-table/categories-table.component.ts]: The modules shouldn't be imported in the component files. 
Remove MatTableModule and MatIconModule from here.

File: src/app/components/categories-table/categories-table.component.ts [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+  valoare: Number = 0;
+  constructor(private endpointsService: EndpointsService, private sanitizer: DomSanitizer, private dialogService: DialogService) {
+  }
+  
+  ngOnInit(): void {
+    this.endpointsService.getCategories().subscribe(categories => {
+      this.dataSource = categories.result;
+      this.createCategoriesToShow();
+    });
+
```
Comment (meda-racasan) [inline src/app/components/categories-table/categories-table.component.ts]: remove the extra new lines

File: api/Controllers/CategoryController.cs [hunk 7]
```diff
@@ -33,6 +33,7 @@ public async Task<IActionResult> GetCategory(int categoryId)
             return Ok(categoryDto);
         }
         
+
```
Comment (meda-racasan) [inline api/Controllers/CategoryController.cs]: remove the extra line

**Pull request merged**